frontend/server/src/Controllers/Course.php

```diff
@@ -485,6 +485,41 @@ private static function resolveGroup(
         return $group;
     }
 
+    /**
+     * @return array{token: string}
+     * @omegaup-request-param mixed $course_alias
+     */
+    public static function apiGenerateTokenForCloneCourse(
+        \OmegaUp\Request $r
+    ): array {
+        if (OMEGAUP_LOCKDOWN) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException('lockdown');
+        }
+
+        $r->ensureMainUserIdentity();
+        \OmegaUp\Validators::validateStringNonEmpty(
+            $r['course_alias'],
+            'course_alias'
+        );
```

Comment:
cortesía de https://github.com/omegaup/omegaup/pull/4562, ya se puede hacer

```suggestion
        $courseAlias = $r->ensureString(
            'course_alias',
            function (string $courseAlias): void {
                \OmegaUp\Validators::validateStringNonEmpty(
                     $r['course_alias'],
                     'course_alias'
                 );
            }
        );
```

en el futuro hay que evitar usar `$r['...']` en cualquier momento, porque mi intención es eliminar eso semi-pronto.

Reply:
Ohh, ese cambio está bueno. Gracias. 

Sólo que tuve que crear `stringNonEmpty` porque no existía, y `validateStringNonEmpty` no hace lo que esperamos aquí. 

Aparte, no sé si realmente se requiera, porque obtengo el siguiente error en psalm:
```
Found a redundant condition when evaluating $courseAlias and trying to reconcile type 'string' to string (see https://psalm.dev/122)
```